### DISCUSSION TITLE
Parameter Tuning

### DISCUSSION THREAD

**Discussion by hamzaakramsybrid** [pyannote/pyannote-audio#732] [Q&A]
Hello to all,
I hope you are fine. 

I have perform speaker diarization on my data using pipeline `torch.hub.load('pyannote/pyannote-audio', 'dia')`  but it was not giving me good results. I explored the speech activity detection and tunned the parameters offset and onset using Binarizae from `pyannote.audio.utils.signal `

```
sad_scores = sad(test_file)
        binarize = Binarize(offset=0.8, onset=0.8, log_scale=True,
                        min_duration_off=0.1, min_duration_on=0.1)
        speech = binarize.apply(sad_scores, dimension=1)
```

and it gave me better results as compare to pipeline. Can anyone please guide me how to tune the parameters in Pipeline as it will improve the speech activity detection which will further improve the speaker diarization.

Thank you so much.

**Comment by hbredin** [comment]
I'd suggest you start with [this tutorial](https://github.com/pyannote/pyannote-audio/blob/master/tutorials/pipelines/speech_activity_detection/README.md) that teaches how to tune pipeline hyper-parameters.

**Comment by hamzaakramsybrid** [comment]
Thank you so much for the suggestion, but I wanted to change the
hyperparameters in Binarize class so I can change the results of speech
activity detection which will further improve the speaker diarizatioin. I
have change tried log_scale False and True and it gave different results.
Also, I have generated results for scale= ['absolute', ;relative',
'percentile] and it is giving different results. The thing that I wanted to
ask is that I have been changing the onset and offset parameters but the
generated results are the same as it is not having any effects by changing
the value of offset and onset. Kindly guide me so I can generate multiple
results by changing the value of onset and offset. Thank you

Regards,
Hamza Akram
